core/src/main/java/org/apache/iceberg/IncrementalDataScan.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Sets;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Set;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.io.CloseableIterable;
+
+class IncrementalDataScan extends DataTableScan implements TableScan {
```

Comment:
I currently delegate to `DataTableScan#planFiles(TableOperations, Snapshot, expressions.Expression, boolean, boolean, Predicate<ManifestFile>, Predicate<ManifestEntry>)` to do the heavy lifting instead of building my own.  
Any pros u see for directly implementing `TableScan` ?

I think we should throw `UnsupportedOperationException` for the methods you mentioned.

Reply:
> Any pros u see for directly implementing TableScan ?

I'm fine with your current approach. I mentioned `TableScan` as it is an interface your code adheres to, so a user could call the above methods, and so we should define their expected behavior.

@rdblue I know we are trying to do a best effort in #497 instead of throwing when the method calls are conflicting, so you ok with throwing `UnsupportedOperationException` here?